### PR TITLE
fix some typos

### PR DESCRIPTION
**Pull request by martinlindhe**
This is the result of me testing out a WIP source code spell-checker
and your project was the random target this time.

### CODE REVIEW

File: src/patterns.rs [hunk 1]
```diff
@@ -118,8 +118,8 @@ impl Rewrite for Pat {
                 };
                 Some(result)
             }
-            PatKind::Struct(ref path, ref fields, elipses) => {
-                rewrite_struct_pat(path, fields, elipses, self.span, context, shape)
+            PatKind::Struct(ref path, ref fields, ellipses) => {
+                rewrite_struct_pat(path, fields, ellipses, self.span, context, shape)
```
Comment (nrc) [inline src/patterns.rs]: This should be `ellipsis`

File: src/patterns.rs [hunk 2]
```diff
@@ -130,7 +130,7 @@ impl Rewrite for Pat {
 fn rewrite_struct_pat(
     path: &ast::Path,
     fields: &[codemap::Spanned<ast::FieldPat>],
-    elipses: bool,
+    ellipses: bool,
```
Comment (nrc) [inline src/patterns.rs]: here and below too

File: tests/source/expr.rs [hunk 3]
```diff
@@ -233,7 +233,7 @@ fn repeats() {
 
 fn blocks() {
     if 1 + 1 == 2 {
-        println!("yay arithmetix!");
+        println!("yay arithmetics!");
```
Comment (nrc) [inline tests/source/expr.rs]: I believe this is deliberate

File: src/patterns.rs (hunk 1 above)
Comment (martinlindhe) [reply src/patterns.rs]: done, thanks!

File: tests/source/expr.rs (hunk 3 above)
Comment (martinlindhe) [reply tests/source/expr.rs]: I removed this change from the PR

File: src/patterns.rs (hunk 2 above)
Comment (martinlindhe) [reply src/patterns.rs]: done

**Comment by martinlindhe** [comment]
i updated the pr and squashed the commits with feedback addressed

File: tests/target/expr.rs [hunk 4]
```diff
@@ -298,7 +298,7 @@ fn repeats() {
 
 fn blocks() {
     if 1 + 1 == 2 {
-        println!("yay arithmetix!");
+        println!("yay arithmetics!");
```
Comment (nrc) [inline tests/target/expr.rs]: THis will need reverting too

Comment (martinlindhe) [reply tests/target/expr.rs]: Thanks, done

**Comment by nrc** [comment]
Thank you!

**Pull request merged**